### PR TITLE
refactor: move back into create react app

### PR DESCRIPTION
**Pull request by rflechtner**
I set up a new project with create-react-app with typescript profile and pulled our code into it.

### CODE REVIEW

**Review by Dudleyneedham — APPROVED** [review]
LGTM

File: src/containers/QuoteCreate/QuoteCreate.tsx [hunk 1]
```diff
@@ -45,15 +45,18 @@ class QuoteCreate extends React.Component<Props, State> {
...
-    const result = {}
-    Object.keys(value.cost.tax).forEach(entryKey => {
+    const result: Record<string, unknown> = {}
+    Object.keys(value.cost.tax).forEach((entryKey) => {
       result[entryKey] = value.cost.tax[entryKey]
     })
     quote.cost.tax = result
-
-    this.setState({ quote })
+    this.setState({ quote: quote as IQuote })
```
Comment (tjwelde) [inline src/containers/QuoteCreate/QuoteCreate.tsx:59]: Is the quote not of type IQuote? How can we check, if it is an IQuote without forcing it?

File: src/containers/IdentitySelector/IdentitySelector.tsx [hunk 2]
```diff
@@ -55,12 +56,15 @@ class IdentitySelector extends React.Component<Props, State> {
...
   }
 
-  private selectIdentity = (selectedOption: SelectIdentityOption): void => {
+  private selectIdentity = (
+    selectedOption: ValueType<SelectIdentityOption>
+  ): void => {
+    if (!selectedOption) return
     const { history, selectIdentity } = this.props
-    if (selectedOption.value === 'create') {
+    if ((selectedOption as SelectIdentityOption).value === 'create') {
```
Comment (tjwelde) [inline src/containers/IdentitySelector/IdentitySelector.tsx:64]: Why does the type need to be enforced here?

**Review by tjwelde — COMMENTED** [review]
Some questions and comemnts

File: src/containers/QuoteCreate/QuoteCreate.tsx (hunk 1 above)
Comment (tjwelde) [reply src/containers/QuoteCreate/QuoteCreate.tsx:59]: the json-schema editor unfortunately isn't generic, so we would have to build a custom type guard, which seems overkill for that

**Pull request merged**